missing go.sum entry for module providing package github.com/shopspring/decimal

_Originally posted by @imkh in https://github.com/prisma/prisma-client-go/issues/402#issuecomment-811171430_

---

Thanks @steebchen! I ran into some issues while updating but after running some commands, I can confirm that it works. I'm not sure if it should be expected to have to run these commands, so here are the logs below. Let me know if you'd prefer a separate issue.

```console
$ go version
go version go1.16.2 darwin/amd64
```

```console
$ ls
go.mod  go.sum  main.go  prisma
$ go get github.com/prisma/prisma-client-go@master
go: downloading github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b
go get: upgraded github.com/prisma/prisma-client-go v0.4.1 => v0.6.1-0.20210330162651-a7eef73a498b
$ go run github.com/prisma/prisma-client-go generate
Prisma schema loaded from prisma/schema.prisma

✔ Generated Prisma Client Go to ./prisma/db in 2.27s

$ go run main.go
../../../../go/pkg/mod/github.com/prisma/prisma-client-go@v0.6.1-0.20210330162651-a7eef73a498b/runtime/types/types.go:10:2: missing go.sum entry for module providing package github.com/shopspring/decimal; to add:
	go mod download github.com/shopspring/decimal
```

```diff
$ git diff
diff --git a/go.mod b/go.mod
index 9a3cbde..a72b26a 100644
--- a/go.mod
+++ b/go.mod
@@ -2,4 +2,4 @@ module demo

 go 1.16

-require github.com/prisma/prisma-client-go v0.4.1 // indirect
+require github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b // indirect
diff --git a/go.sum b/go.sum
index 601ab22..552236a 100644
--- a/go.sum
+++ b/go.sum
@@ -4,6 +4,9 @@ github.com/iancoleman/strcase v0.0.0-20190422225806-e506e3ef7365/go.mod h1:SK73t
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/prisma/prisma-client-go v0.4.1 h1:Uad0Gu21wbhLbf7VJLwTh/LEBufC53BGnaLVGY0V5rQ=
 github.com/prisma/prisma-client-go v0.4.1/go.mod h1:N2XDbo3cYql1O0EjheOvS1q5g6vIlLzvNILdFUuYI/k=
+github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b h1:ooz+z7xDWTeL0T5Y7zr4v/UbPk97OpyOJhNelHPX4Gw=
+github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b/go.mod h1:p08hGN9qp6hxJXkxfO+xbAjBMwuNXbUN+slBhuPUEi0=
+github.com/shopspring/decimal v1.2.0/go.mod h1:DKyhrW/HYNuLGql+MJL6WCR6knT2jwCFRcu2hWCYk4o=
 github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
 github.com/takuoki/gocase v1.0.0 h1:gPwLJTWVm2T1kUiCsKirg/faaIUGVTI0FA3SYr75a44=
```

---

The problem is fixed with either:

- `go mod download github.com/shopspring/decimal`

```console
$ go mod download github.com/shopspring/decimal # No changes in go.sum
$ go run main.go # It works
```

- or `go mod tidy`

```console
$ go mod tidy # Changes in go.sum
go: downloading github.com/shopspring/decimal v1.2.0
go: downloading github.com/stretchr/testify v1.4.0
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/davecgh/go-spew v1.1.0
go: downloading github.com/pmezard/go-difflib v1.0.0
```
```diff
$ git diff
diff --git a/go.mod b/go.mod
index a72b26a..bf2960b 100644
--- a/go.mod
+++ b/go.mod
@@ -2,4 +2,4 @@ module demo

 go 1.16

-require github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b // indirect
+require github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b
diff --git a/go.sum b/go.sum
index d4cddfd..2a21397 100644
--- a/go.sum
+++ b/go.sum
@@ -1,13 +1,16 @@
+github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
-github.com/iancoleman/strcase v0.0.0-20190422225806-e506e3ef7365 h1:ECW73yc9MY7935nNYXUkK7Dz17YuSUI9yqRqYS8aBww=
 github.com/iancoleman/strcase v0.0.0-20190422225806-e506e3ef7365/go.mod h1:SK73tn/9oHe+/Y0h39VT4UCxmurVJkR5NA7kMEAOgSE=
+github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b h1:ooz+z7xDWTeL0T5Y7zr4v/UbPk97OpyOJhNelHPX4Gw=
 github.com/prisma/prisma-client-go v0.6.1-0.20210330162651-a7eef73a498b/go.mod h1:p08hGN9qp6hxJXkxfO+xbAjBMwuNXbUN+slBhuPUEi0=
+github.com/shopspring/decimal v1.2.0 h1:abSATXmQEYyShuxI4/vyW3tV1MrKAJzCZ/0zLUXYbsQ=
 github.com/shopspring/decimal v1.2.0/go.mod h1:DKyhrW/HYNuLGql+MJL6WCR6knT2jwCFRcu2hWCYk4o=
 github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
+github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJyk=
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
-github.com/takuoki/gocase v1.0.0 h1:gPwLJTWVm2T1kUiCsKirg/faaIUGVTI0FA3SYr75a44=
 github.com/takuoki/gocase v1.0.0/go.mod h1:QgOKJrbuJoDrtoKswBX1/Dw8mJrkOV9tbQZJaxaJ6zc=
 gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
+gopkg.in/yaml.v2 v2.2.2 h1:ZCJp+EgiOT7lHqUV2J862kp8Qj64Jo6az82+3Td9dZw=
 gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
```
```console
$ go run main.go # It works
```

The error does not occur with v0.6.0 so it was introduced in the [few commits](https://github.com/prisma/prisma-client-go/compare/v0.6.0...a7eef73a498b1863e19365203e90c49649e411c5) since that release.

--- a/generator/templates/_header.gotpl
+++ b/generator/templates/_header.gotpl
@@ -16,6 +16,11 @@ import (
 	"github.com/prisma/prisma-client-go/runtime/raw"
 	"github.com/prisma/prisma-client-go/runtime/builder"
 	"github.com/prisma/prisma-client-go/runtime/transaction"
+
+	// no-op import for go modules
+	_ "github.com/shopspring/decimal"
+	_ "github.com/iancoleman/strcase"
+	_ "github.com/takuoki/gocase"
 )
 
 // re-declare variables which are needed in Prisma Client Go but also should be exported

--- a/runtime/lifecycle/lifecycle.go
+++ b/runtime/lifecycle/lifecycle.go
@@ -8,7 +8,7 @@ type Lifecycle struct {
 	Engine engine.Engine
 }
 
-// Connects to the Prisma query engine. Required to call before accessing data.
+// Connect connects to the Prisma query engine. Required to call before accessing data.
 // It is recommended to immediately defer calling Disconnect.
 //
 // Example:
@@ -26,7 +26,7 @@ func (c *Lifecycle) Connect() error {
 	return c.Engine.Connect()
 }
 
-// Disconnects from the Prisma query engine.
+// Disconnect disconnects from the Prisma query engine.
 // This is usually invoked on kill signals in long running applications (like webservers),
 // or when no database access is needed anymore (like after executing a CLI command).
 //

